### PR TITLE
nrf5-sdk: init at 17.1.0

### PR DESCRIPTION
**Pull request by StarGate01**
###### Description of changes

This PR adds the Nordic Semiconductor [nRF5 SDK](https://www.nordicsemi.com/Products/Development-software/nRF5-SDK) to `/share/nRF5_SDK` .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/nrf5-sdk/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  nativeBuildInputs = [ unzip ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  unpackPhase = ''
+    unzip $src
+  '';
+
+  installPhase = ''
```
Comment (legendofmiracles) [inline pkgs/development/libraries/nrf5-sdk/default.nix:28]: Please run the preInstall and postInstall phases with `runHook`

File: pkgs/development/libraries/nrf5-sdk/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+, unzip
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nrf5-sdk";
+  version = "17.1.0";
+
+  urlHash = "ddde560";
+
+  src = fetchurl {
```
Comment (legendofmiracles) [inline pkgs/development/libraries/nrf5-sdk/default.nix:13]: I believe that if you use fetchzip instead you won’t have to bother with unpacking it manually

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/nrf5-sdk/default.nix (hunk 2 above)
Comment (StarGate01) [reply pkgs/development/libraries/nrf5-sdk/default.nix:13]: I tried doing that, but it failed with `do not know how to unpack source archive`.

Comment (legendofmiracles) [reply pkgs/development/libraries/nrf5-sdk/default.nix:13]: I've written this [small patch](http://0x0.st/oSBw.txt) to fix it

Comment (StarGate01) [reply pkgs/development/libraries/nrf5-sdk/default.nix:13]: Thabk you! I will apply it later today.

Comment (StarGate01) [reply pkgs/development/libraries/nrf5-sdk/default.nix]: @legendofmiracles You patch appears to function correctly. I suspect I had some weird caching or network stability issues before.

**Pull request merged**